### PR TITLE
VMHostVMKernelModule, VMHostSNMPAgent, VMHostSoftwareDevice and VMHostSharedSwapSpace DSC Resources

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Added VMHostVMKernelModule DSC Resource.
- Added VMHostSNMPAgent DSC Resource.
- Added VMHostSoftwareDevice DSC Resource.
- Added VMHostSharedSwapSpace DSC Resource.
- Added Unit Tests for VMHostVMKernelModule DSC Resource.
- Added Unit Tests for VMHostSNMPAgent DSC Resource.
- Added Unit Tests for VMHostSoftwareDevice DSC Resource.
- Added Unit Tests for VMHostSharedSwapSpace DSC Resource.
- Added Integration Tests for VMHostVMKernelModule DSC Resource.
- Added Integration Tests for VMHostSNMPAgent DSC Resource.
- Added Integration Tests for VMHostSoftwareDevice DSC Resource.
- Added Integration Tests for VMHostSharedSwapSpace DSC Resource.
- Added Documentation and example Configuration for VMHostVMKernelModule DSC Resource.
- Added Documentation and example Configurations for VMHostSNMPAgent DSC Resource.
- Added Documentation and example Configurations for VMHostSoftwareDevice DSC Resource.
- Added Documentation and example Configuration for VMHostSharedSwapSpace DSC Resource.

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Example usage of VMHostVMKernelModule DSC Resource:

```
VMHostVMKernelModule VMHostVMKernelModule {
    Server = $Server
    Credential = $Credential
    Name = $VMHostName
    Module = $VMKernelModule
    Enabled = $true
    Force = $true
}
```

**Comment by SimeonGerginov** [comment]
Example usage of VMHostSNMPAgent DSC Resource:

```
VMHostSNMPAgent VMHostSNMPAgent {
    Server = $Server
    Credential = $Credential
    Name = $VMHostName
    Module = $VMKernelModule
    Authentication = 'SHA1'
    Enable = $true
    Hwsrc = 'indications'
    LargeStorage = $true
    LogLevel = 'info'
    Port = 161
    Privacy = 'AES128'
}
```

**Comment by SimeonGerginov** [comment]
Example usage of VMHostSoftwareDevice DSC Resource:

```
VMHostSoftwareDevice VMHostSoftwareDevice {
    Server = $Server
    Credential = $Credential
    Name = $VMHostName
    DeviceIdentifier = $DeviceId
    Ensure = 'Present'
    InstanceAddress = 0
}
```

**Comment by SimeonGerginov** [comment]
Example usage of VMHostSharedSwapSpace DSC Resource:

```
VMHostSharedSwapSpace VMHostSharedSwapSpace {
    Server = $Server
    Credential = $Credential
    Name = $VMHostName
    DatastoreEnabled = $true
    DatastoreName = $DatastoreName
    DatastoreOrder = 1
    HostCacheEnabled = $false
    HostCacheOrder = 2
    HostLocalSwapEnabled = $false
    HostLocalSwapOrder = 0
}
```

File: Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Parameters
+
+| Parameter | Attribute | DataType | Description | Allowed Values |
+| --- | --- | --- | --- | --- |
+| **Server** | Key | string | The name of the Server we are trying to connect to. The Server can be a vCenter or ESXi. ||
+| **Credential** | Mandatory | PSCredential | The credentials needed for connection to the specified Server. ||
+| **Name** | Key | string | The name of the VMHost. ||
+| **DatastoreEnabled** | Optional | bool | Specifies if the Datastore option should be enabled or not. ||
+| **DatastoreName** | Optional | string | The name of the Datastore used by the Datastore option. ||
+| **DatastoreOrder** | Optional | long | The order of the Datastore option in the preference of the options. ||
```
Comment (dmilov) [inline Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md]: It's not clear to me what this order is? Single datastore is specified with DatastoreName, what means order? Update the description

File: Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+| **HostCacheEnabled** | Optional | bool | Specifies if the host cache option should be enabled or not. ||
+| **HostCacheOrder** | Optional | long | The order of the host cache option in the preference of the options. ||
```
Comment (dmilov) [inline Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md]: Same as above. Probably the meaning of order has to be described better for all properties

File: Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md (hunk 1 above)
Comment (SimeonGerginov) [reply Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md]: Fixed.

File: Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md (hunk 2 above)
Comment (SimeonGerginov) [reply Documentation/DSCResources/VMHostSharedSwapSpace/VMHostSharedSwapSpace.md]: Fixed.

**Pull request merged**